sklearn/model_selection/tests/test_search.py

```diff
@@ -591,71 +552,212 @@ def test_param_sampler():
         assert_equal([x for x in sampler], [x for x in sampler])
 
 
-def test_randomized_search_grid_scores():
+def check_results_array_types(results, param_keys, score_keys):
+    # Check if the search results' array are of correct types
+    assert_true(all(isinstance(results[param], np.ma.MaskedArray)
+                    for param in param_keys))
+    assert_true(all(results[key].dtype is np.dtype('O')
+                    for key in param_keys))
+    assert_false(any(isinstance(results[key], np.ma.MaskedArray)
+                     for key in score_keys))
+    assert_true(all(results[key].dtype is np.dtype('float64')
+                    for key in score_keys if key != 'test_rank_score'))
```

Comment:
Please use `results[key].dtype == np.float64` instead of relying on the physical equality and dtype interning tricks.